### PR TITLE
Invideo quiz

### PR DESCRIPTION
**Pull request by itaykinnrot**

### CODE REVIEW

File: modules/Quiz/resources/AddQ_hover.svg [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (itaykinnrot) [inline modules/Quiz/resources/AddQ_hover.svg]: @mrkfdr  please move all non js files to images folder under resouces

File: modules/Quiz/resources/bigButton.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.bigButton {
+	cursor: pointer;
```
Comment (itaykinnrot) [inline modules/Quiz/resources/bigButton.css]: @mrkfdr  please move to css folder

File: modules/Quiz/resources/bigButton.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+					if( newState == 'load' ){
+						_this.hide(true);
+					}
+					if( newState == 'pause' && _this.getPlayer().isPauseLoading ) {
+						_this.hide();
+					}
+					if (newState == 'start') {
+						_this.hide(true);
+					}
+                    if (newState == 'play') {
```
Comment (itaykinnrot) [inline modules/Quiz/resources/bigButton.js]: @mrkfdr  please fix the indent

File: modules/Quiz/resources/bigButton.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
+( function( mw, $ ) {"use strict";
+	mw.PluginManager.add( 'bigButton', mw.KBaseComponent.extend({
```
Comment (itaykinnrot) [inline modules/Quiz/resources/bigButton.js]: @mrkfdr please change the plugin name - this is too generic

File: modules/Quiz/resources/bigButton.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+				}
+			},
+			onDisable: function(){
+				this.isDisabled = true;
+				this.hideComponent();
+			},
+			getComponent: function() {
+				var _this = this;
+				var eventName = 'click';
+				if (mw.isMobileDevice()){
```
Comment (itaykinnrot) [inline modules/Quiz/resources/bigButton.js]: please change to if ( mw.isAndroid() ){

File: modules/Quiz/resources/quiz.css [hunk 6]
```diff
@@ -0,0 +1,571 @@
+@font-face {
+    font-family: 'LatoLight';
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.css]: move to css folder

File: modules/Quiz/resources/quiz.js [hunk 7]
```diff
@@ -0,0 +1,917 @@
+(function (mw, $) {
+    "use strict";
+    $.cpObject = {};
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: why do you use this method?

File: modules/Quiz/resources/quiz.js [hunk 8]
```diff
@@ -0,0 +1,917 @@
...
+            }, {
+                'service': 'quiz_quiz',
+                'action': 'get',
+                'entryId': _this.embedPlayer.kentryid
+            }
+
+            ];
+
+            _this.getKClient().doRequest(getQuizuserEntryIdAndQuizParams, function (data) {
+                console.log("Quiz Params -->");
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: remove console - use this.log or mw.log

File: modules/Quiz/resources/quiz.js [hunk 9]
```diff
@@ -0,0 +1,917 @@
...
+                console.log(data);
+
+                $.grep(data, function (e) {
+
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: remove extra line

File: modules/Quiz/resources/quiz.js [hunk 10]
```diff
@@ -0,0 +1,917 @@
...
+                    if ( data[1].uiAttributes === undefined ){
+                        console.log('Connect to quiz err -->', e.code, e.message);
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: remove console

File: modules/Quiz/resources/quiz.js [hunk 11]
```diff
@@ -0,0 +1,917 @@
...
+
+            ];
+
+            _this.getKClient().doRequest(getQuizuserEntryIdAndQuizParams, function (data) {
+                console.log("Quiz Params -->");
+                console.log(data);
+
+                $.grep(data, function (e) {
+
+                    if ( data[1].uiAttributes === undefined ){
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: make sure that data is an array before to avoid error

File: modules/Quiz/resources/quiz.js [hunk 12]
```diff
@@ -0,0 +1,917 @@
...
+                'filter:orderBy': '-createdAt'
+            }, {
+                'service': 'quiz_quiz',
+                'action': 'get',
+                'entryId': _this.embedPlayer.kentryid
+            }
+
+            ];
+
+            _this.getKClient().doRequest(getQuizuserEntryIdAndQuizParams, function (data) {
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: where is the failed case?

File: modules/Quiz/resources/quiz.js [hunk 13]
```diff
@@ -0,0 +1,917 @@
...
+                        _this._errMsg();
+                        return false;
+                    }
+                });
+
+                $.quizParams = data[1];
+
+                if (data[0].totalCount > 0 &&  !$.isEmptyObject(data[0].objects[0])) {
+                    _this.kQuizUserEntryId = data[0].objects[0].id;
+                    console.log('exist kQuizUserEntryId-----> ' +_this.kQuizUserEntryId);
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: replace console

File: modules/Quiz/resources/quiz.js [hunk 14]
```diff
@@ -0,0 +1,917 @@
...
+                }
+                else {
+                    var createQuizuserEntryId = {
+                        'service': 'userEntry',
+                        'action': 'add',
+                        'userEntry:objectType': 'KalturaQuizUserEntry',
+                        'userEntry:entryId': _this.embedPlayer.kentryid
+                    };
+
+                    _this.getKClient().doRequest(createQuizuserEntryId, function (data) {
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: where is the failed case?

File: modules/Quiz/resources/quiz.js [hunk 15]
```diff
@@ -0,0 +1,917 @@
...
+                        console.log('create user entry Id->');
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: remove console

File: modules/Quiz/resources/quiz.js [hunk 16]
```diff
@@ -0,0 +1,917 @@
...
+                        console.log(data);
+                        //if (data.objectType==='KalturaAPIException'){
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: remove this code..

File: modules/Quiz/resources/quiz.js [hunk 17]
```diff
@@ -0,0 +1,917 @@
...
+                _this.isSeekingIVQ = true;
+            });
+
+            embedPlayer.addJsListener( 'kdpReady', function(){
+                if (embedPlayer.autoplay) embedPlayer.autoplay = false;
+                embedPlayer.removeJsListener( 'kdpReady');
+            });
+            embedPlayer.addJsListener( 'playerPlayEnd', function(){
+                var anUnswered = _this.getUnansweredQuestNrs();
+                if (anUnswered) _this.almostDone(anUnswered);
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: please add {}

File: modules/Quiz/resources/quiz.js [hunk 18]
```diff
@@ -0,0 +1,917 @@
...
+            var $template = $(quizStartTemplate({
+                'quiz': this,
+                quizStartTemplate: quizStartTemplate
+            }));
+            return defer.resolve($template);
+        },
+        _continuePlay: function () {
+            var _this = this, embedPlayer = this.getPlayer();
+            if (!_this.isErr) {
+                if (_this.isScreenVisible())_this.removeScreen();
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: add {}

File: modules/Quiz/resources/quiz.js [hunk 19]
```diff
@@ -0,0 +1,917 @@
...
+                embedPlayer.enablePlayControls();
+                embedPlayer.play();
+                _this.displayBubbles();
+            }
+        },
+        _initParams: function () {
+            var _this = this;
+            $.grep($.quizParams.uiAttributes, function (e) {
+
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: remove extra line

File: modules/Quiz/resources/quiz.js [hunk 20]
```diff
@@ -0,0 +1,917 @@
...
+                    .on('click', function () {
+                    _this.checkIfDone(0);
+                    });
+
+            });
+        },
+
+       _checkCuepointsReady:function(callback){
+           var intrVal;
+           intrVal = setInterval(function () {
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: please assign setInterval to global(in this class) var and make sure that you dont setInterval that already started

File: modules/Quiz/resources/quiz.js [hunk 21]
```diff
@@ -0,0 +1,917 @@
...
+        },
+
+        _gotoScrubberPos: function (questionNr) {
+            var player = this.getPlayer();
+            player.sendNotification('doSeek', ($.cpObject.cpArray[questionNr].startTime) / 999);
+        },
+        qCuePointHandler: function (e, cuePointObj) {
+            var _this = this;
+
+            if (!$.quizParams.showCorrectAfterSubmission && _this.score !== null) return;
```
Comment (itaykinnrot) [inline modules/Quiz/resources/quiz.js]: add {}

File: modules/Quiz/resources/quiz.js (hunk 7 above)
Comment (mrkfdr) [inline modules/Quiz/resources/quiz.js]: Yes to store/update/check/display all cuepoints data

File: modules/Quiz/resources/bigButton.js [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+			},
+			hideComponent: function( force ) {
+				if( force ) {
+					this.getComponent().hide();
+				}
+			},
+			clickButton: function( event ){
+				event.preventDefault();
+				event.stopPropagation();
+				this.getPlayer().sendNotification('big_button_clicked', this.embedPlayer.currentTime);
```
Comment (itaykinnrot) [inline modules/Quiz/resources/bigButton.js]: @atarsh please move the event name to configuration

File: modules/Quiz/resources/bigButton.css (hunk 2 above)
Comment (atarsh) [inline modules/Quiz/resources/bigButton.css]: done

File: modules/Quiz/resources/bigButton.js (hunk 4 above)
Comment (atarsh) [inline modules/Quiz/resources/bigButton.js]: moved to new module - CustomButton

File: modules/Quiz/resources/bigButton.js (hunk 3 above)
Comment (atarsh) [inline modules/Quiz/resources/bigButton.js]: done

File: modules/Quiz/resources/bigButton.js (hunk 5 above)
Comment (atarsh) [inline modules/Quiz/resources/bigButton.js]: done

File: modules/Quiz/resources/bigButton.js (hunk 22 above)
Comment (atarsh) [inline modules/Quiz/resources/bigButton.js]: done

**Pull request closed without merging**